cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/AbstractGoal.java

```diff
@@ -309,6 +312,113 @@ Replica maybeApplySwapAction(ClusterModel clusterModel,
     return null;
   }
 
+  /**
+   * Attempt to move replica between disks of the same broker. The application considers the candidate disks as the potential
+   * destination disk for replica movement. If the movement attempt succeeds, the function returns the destination disk,
+   * otherwise the function returns null.
+   *
+   * @param clusterModel    The state of the cluster.
+   * @param replica         Replica to be moved.
+   * @param candidateDisks  Candidate disks as the potential destination for replica movement.
+   * @param optimizedGoals  Optimized goals.
+   * @return The destination disk if the movement attempt succeeds, null otherwise.
+   */
+  protected Disk maybeMoveReplicaBetweenDisks(ClusterModel clusterModel,
+                                              Replica replica,
+                                              Collection<Disk> candidateDisks,
+                                              Set<Goal> optimizedGoals) {
+    for (Disk disk : candidateDisks) {
+      BalancingAction proposal = new BalancingAction(replica.topicPartition(),
```

Comment:
Can we update the `JavaDoc` of `BalancingAction` to indicate when we expect the `destinationTp` to be `null`?